front-end/src/components/CustomNode/GraphView.js

```diff
@@ -32,109 +33,124 @@ export default class GraphView extends React.Component {
+                data: json,
+                columns: columns,
+                rows: rows,
+                loading: false,
+                widths: widths
+            });
+          })
+          .catch(err => console.error(err));
+  };
 
     Cell = ({ columnIndex, rowIndex, style }) => {
-      const className =  columnIndex % 2
-          ? rowIndex % 2 === 0
-            ? 'GridItemOdd'
-            : 'GridItemEven'
-          : rowIndex % 2
-            ? 'GridItemOdd'
-            : 'GridItemEven';
-
-      if (rowIndex === 0) {
-        return (
-          <div className={className} style={style}>
-            {this.state.keys[columnIndex]}
-          </div>
-        );
-      }
+      const className = (rowIndex % 2 === 0) ? 'GridItemEven' : 'GridItemOdd';
+      const column = this.state.columns[columnIndex];
 
       return (
         <div className={className} style={style}>
-          {this.state.data[this.state.keys[columnIndex]][rowIndex.toString()] }
+          {(rowIndex === 0) ? column : this.state.data[column][rowIndex.toString()]}
         </div>
       );
-    }
+    };
 
 
     render() {
+      let body;
+      let footer;
+
       if (this.state.loading) {
-        return (<div>Loading data...</div>);
+          // Print loading spinner
+          body = (<Roller color="black" />);
+      } else if (this.state.data.length < 1) {
+          // Print instructions about loading
+          body = "Loading the data might take a while depending on how big the data is.";
+          footer = (
+              <Modal.Footer>
+                <Button variant="secondary" onClick={this.onClose}>Cancel</Button>
+                <Button variant="secondary"
+                        disabled={this.props.node.options.status !== "complete"}
+                        onClick={this.load}>Load
+                </Button>
+              </Modal.Footer>
+          );
+      } else {
+          // Display the grid
+          let displayHeight = this.state.rows.length * 20;
+          let displayWidth = this.state.maxWidth;
+
+          body = (
+              <Grid
+                  ref={this.state.gridRef}
+                  className="Grid"
+                  columnCount={this.state.columns.length}
+                  columnWidth={index => this.columnWidths(index)}
+                  height={displayHeight < 600 ? displayHeight + 5 : 600}
+                  rowCount={this.state.rows.length}
+                  rowHeight={index => 20}
+                  width={displayWidth < 1000 ? displayWidth : 1000}
+              >
+                {this.Cell}
+              </Grid>
+          );
       }
 
-      if (this.state.columnCount < 1) {
-        return (
-          <Modal show={this.props.show} onHide={this.props.toggleShow} centered
-             onWheel={e => e.stopPropagation()}>
-          <Modal.Header>
-              <Modal.Title><b>{this.props.node.options.name}</b> View</Modal.Title>
+      return (
+          <Modal
```

Comment:
Using the `dialogClassName` prop should allow you style this modal and its children specifically

Reply:
Exactly what I was looking for, thanks! Fixed in 5b593e4.